lib/ts/recipe/emailpassword/components/signInAndUp/api.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2020, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Imports.
+ */
+
+import { INCORRECT_EMAIL_PASSWORD_COMBINATION_ERROR, SOMETHING_WENT_WRONG_ERROR } from "../../../../constants";
+import { APIFormField } from "../../../../types";
+import { API_RESPONSE_STATUS } from "../../constants";
+import { SignInThemeResponse, SignUpAPI, SignInAPI, SignUpThemeResponse } from "../../types";
+
+/*
+ * Methods.
+ */
+
+export async function handleSignUpAPI(
+    formFields: APIFormField[],
+    rid: string,
+    signUpAPI: SignUpAPI
+): Promise<SignUpThemeResponse> {
+    try {
+        const headers: HeadersInit = {
+            rid
+        };
+        const response = await signUpAPI({ formFields }, headers);
+
+        // Otherwise, if field errors.
+        if (response.status === API_RESPONSE_STATUS.FIELD_ERROR) {
+            return {
+                status: API_RESPONSE_STATUS.FIELD_ERROR,
+                formFields: response.formFields
+            };
+        }
+
+        // Otherwise, if success.
+        if (response.status === API_RESPONSE_STATUS.OK) {
+            return {
+                ...response
+            };
+        }
+
+        // Otherwise, something went wrong.
+        return {
+            status: API_RESPONSE_STATUS.GENERAL_ERROR,
+            message: SOMETHING_WENT_WRONG_ERROR
+        };
```

Comment:
Right, I wasn't sure if the API was returning `GENERAL_ERROR` in some cases or if it was only for status code > 300.
I've refactored so that API response type does not include GENERAL_ERROR and theme response types includes it.

Reply:
According to the FDI spec, there should never be a status of `GENERAL_ERROR` at all returned by the API.